src/main/java/io/strimzi/kafka/metrics/KafkaMetricsCollector.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.metrics;
+
+import io.prometheus.client.Collector;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.KafkaMetric;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+/**
+ * Prometheus Collector to store and export metrics retrieved by the reporters.
+ */
+public class KafkaMetricsCollector extends Collector {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaMetricsCollector.class.getName());
+
+    private final Map<MetricName, KafkaMetric> metrics;
+    private final PrometheusMetricsReporterConfig config;
+    private String prefix;
+
+    public KafkaMetricsCollector(PrometheusMetricsReporterConfig config) {
+        this.config = config;
+        this.metrics = new ConcurrentHashMap<>();
+    }
+
+    public void setPrefix(String prefix) {
+        this.prefix = prefix;
+    }
+
+    @Override
+    public List<MetricFamilySamples> collect() {
+        List<MetricFamilySamples> samples = new ArrayList<>();
+
+        for (Map.Entry<MetricName, KafkaMetric> entry : metrics.entrySet()) {
+            MetricName metricName = entry.getKey();
+            KafkaMetric kafkaMetric = entry.getValue();
+            LOG.trace("Collecting Kafka metric {}", metricName);
+
+            String name = metricName(metricName);
```

Comment:
Is there a reason we can't compute the name when inserting and removing metrics from the map, rather than here (which I assume is a hotter path)? If there is a reason it would be worth a comment.

Reply:
The only reason it's here is to follow the same logic as in `YammerMetricsCollector`. But yeah for this code path this logic should be done when adding the metric instead of recomputing it each time the metrics are collected.